bindings/dotnetcore/dotnet-core-binding/Microsoft.Azure.Devices.Gateway/Gateway.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Azure.Devices.Gateway
+{
+    /// <summary>
+    ///  Wrapper for Azure IoT Gateway SDK. 
+    /// </summary>
+    public class Gateway
```

Comment:
This isn't a class in the OO sense, it's just a direct translation of the C API (including interop types like `IntPtr`), bundled as a group of static methods. So maybe it should have a more utilitarian name. If we ever come back and implement a C#-ish class for gateway, we would use a prettier name.

```csharp
using Microsoft.Azure.Devices.Gateway;
IntPtr gateway = Interop.Gateway_CreateFromJson("path/to/config");
Interop.Gateway_Destroy(gateway);
```